### PR TITLE
Adding v4 -> v5 upgrade chain test

### PR DESCRIPTION
**Pull request by chatton**
## Description

This PR adds a test which for now uses a hard coded version of v4 and v5 (from [this pr](https://github.com/cosmos/ibc-go/pull/2144))

We can update this to use the next v5 rc tag when the changes are included.

part of: #1908

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/tests/upgrades/upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	height, err = chain.Height(ctx)
+	s.Require().NoError(err, "error fetching height after upgrade")
+
+	s.Require().GreaterOrEqual(height, haltHeight+blocksAfterUpgrade, "height did not increment enough after upgrade")
+}
+
+func (s *UpgradeTestSuite) TestV4ToV5ChainUpgrade() {
+	// TODO: temporarily hard code the version upgrades.
+	oldVersion := "v4.0.0"
+	targetVersion := "pr-2144" // v5 version with upgrade handler, replace with v5.0.0-rc3 when it is cut.
```
Comment (colin-axner) [inline e2e/tests/upgrades/upgrade_test.go:78]: I guess we would need to backport upgrade handlers for each version we want to test?

File: e2e/tests/upgrades/upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	s.Require().NoError(os.Setenv(testconfig.ChainATagEnv, oldVersion))
+	s.Require().NoError(os.Setenv(testconfig.ChainBTagEnv, oldVersion))
+
+	ctx := context.Background()
+
+	s.SetupChainsRelayerAndChannel(ctx)
+	chainA, _ := s.GetChains()
+
+	chainAUser := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	s.UpgradeChain(ctx, chainA, chainAUser, "normal upgrade", targetVersion)
```
Comment (colin-axner) [inline e2e/tests/upgrades/upgrade_test.go:88]: I guess followup pr's can allow pre/post upgrade checks

**Review by colin-axner — APPROVED** [review]
Wahoo!

File: e2e/tests/upgrades/upgrade_test.go (hunk 1 above)
Comment (chatton) [reply e2e/tests/upgrades/upgrade_test.go:78]: yep we won't be able to upgrade anything unless we have them.

File: e2e/tests/upgrades/upgrade_test.go (hunk 2 above)
Comment (chatton) [reply e2e/tests/upgrades/upgrade_test.go:88]: Yes I think we can add before/after transfers in a follow up.

**Review by damiannolan — APPROVED** [review]
Awesome, nice work! This will be so valuable

**Pull request merged**